detectron2/data/datasets/coco.py

```diff
@@ -280,6 +287,117 @@ def file2id(folder_path, file_path):
     return dataset_dicts
 
 
+def convert_to_coco_dict(dataset_name):
+    """
+    Convert a generic dataset into COCO json format
+
+    Generic dataset description can be found here:
+    https://github.com/facebookresearch/detectron2/blob/master/docs/tutorials/datasets.md#register-a-dataset
+
+    COCO data format description can be found here:
+    http://cocodataset.org/#format-data
+
+    Args:
+        dataset_name: name of the source dataset
+    Returns:
+        coco_dict: serializable dict in COCO json format
+    """
+
+    dataset_dicts = DatasetCatalog.get(dataset_name)
+    categories = [
+        {"id": id, "name": name}
+        for id, name in enumerate(MetadataCatalog.get(dataset_name).thing_classes)
+    ]
+
+    logger.info("Converting dataset dicts into COCO format")
+    images = []
+    annotations = []
+
+    # just for logging purposes
+    _annotation_keys = Counter()
+
+    for image_dict in dataset_dicts:
+        image = {
+            "id": image_dict["image_id"],
+            "width": image_dict["width"],
+            "height": image_dict["height"],
+            "file_name": image_dict["file_name"],
+        }
+
+        images.append(image)
+
+        # deep-copying various annotations from the original format
+        # can be bbox, segmentation, keypoint, etc.
+        anns_per_image = deepcopy(image_dict["annotations"])
+        for annotation in anns_per_image:
+            # COCO requirement: linking annotations to images
+            annotation["id"] = len(annotations) + 1
+            annotation["image_id"] = image_dict["image_id"]
+
+            # COCO requirement: XYWH box format
+            bbox = annotation["bbox"]
+            bbox_mode = annotation["bbox_mode"]
+            bbox = BoxMode.convert(bbox, bbox_mode, BoxMode.XYWH_ABS)
+            del annotation["bbox_mode"]
```

Comment:
I'm deleting "bbox_mode" because it is not serializable - as mentioned in L342 - and not because it is an optional field. I completely agree with you on keeping the extra keys, as long as they can be saved to and loaded from json files.

Reply:
Understood, done in [6d6159](https://github.com/facebookresearch/detectron2/pull/175/commits/6d615904c33fd1f490cb8047de925663329e7065)